config/orte_setup_debugger_flags.m4

```diff
@@ -53,4 +53,26 @@ AC_DEFUN([ORTE_SETUP_DEBUGGER_FLAGS],[
 
     AC_SUBST(CFLAGS_WITHOUT_OPTFLAGS)
     AC_SUBST(DEBUGGER_CFLAGS)
+
+    # Check for compiler specific flag to add in unwind information.
+    # This is needed when attaching using MPIR to unwind back to the
+    # user's main function. Certain optimisations can prevent GDB from
+    # producing a stack when explicit unwind information is unavailable.
+    # This is implied by -g, but we want to save space and don't need
+    # full debug symbols.
+    if test "x$opal_cv_c_compiler_vendor" = "xibm"; then
+        MPIR_UNWIND_CFLAGS="-g"
+    elif test "x$opal_cv_c_compiler_vendor" = "xgnu"; then
+        MPIR_UNWIND_CFLAGS="-fasynchronous-unwind-tables"
+    elif test "x$opal_cv_c_compiler_vendor" = "xintel"; then
+        MPIR_UNWIND_CFLAGS="-fasynchronous-unwind-tables"
+    elif test "x$opal_cv_c_compiler_vendor" = "xportland group"; then
+        MPIR_UNWIND_CFLAGS="-Meh_frame -Mframe"
+    else
+        MPIR_UNWIND_CFLAGS=
+    fi
```

Comment:
It might be better to actually try the flags in question, rather than assume that they work for a given compiler vendor.  E.g., perhaps something (warning: typed directly into github -- might not be 100% correct):

```
dnl Check to see if specific CFLAGS work
dnl $1: compiler flags to check
dnl $2: Action if the flags work
dnl $3: Action if the flags do not work
AC_DEFUN([_ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS],[
    OPAL_VAR_SCOPE_PUSH([ORTE_SETUP_DEBUGGER_FLAGS_CFLAGS_save])
    ORTE_SETUP_DEBUGGER_FLAGS_CFLAGS_save=$CFLAGS
    AC_MSG_CHECKING([if $1 compiler flag works])
    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[int i = 3;])],
        [$2; AC_MSG_RESULT([yes])],
        [$3; AC_MSG_RESULT([no])])
    CFLAGS=$ORTE_SETUP_DEBUGGER_FLAGS_CFLAGS_save
    OPAL_VAR_SCOPE_POP
])
```

Then where you have this current `if` block checking compiler vendors, instead do something like:

```
    _ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-fasynchronous-unwind-tables],
        [MPIR_UNWIND_CFLAGS=-fasynchronous-unwind-tables],
        [_ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-Meh_frame -Mframe],
            [MPIR_UNWIND_CFLAGS="-Meh_frame -Mframe"],
            [MPIR_UNWIND_CFLAGS=-g])
        ])
```

Waving my hands here a little -- try something along these lines (it doesn't have to be exactly this).  My point is the assuming values based on checking vendors and/or versions is inherently imprecise.  It's better to actually just test for the behavior and see if it works.